File: projects/m2/020-write-out-numbers-in-english/solution/jimmyhu/solution-write-out-numbers-in-english.py
def wordnumber(num):
    labl = {
        'unit': ['one','two','three','four','five','six','seven','eight','nine'],
        'special': ['eleven','twelve','thirteen','fourteen'],
        'decim': ['twenty','thirty','forty',],
        'prefix': ['fif','six','seven','eigh','nine'],
        'suffix':['teen','ty'],
    }
    target = 2
    if len(num) < 3:
        target -=1
    if len(num) > 1:
        if num[target-1] == '1':
            if int(num[target]) > 4:
                decimals = labl['prefix'][int(num[target])-5] + labl['suffix'][0]
            else:
                decimals = labl['special'][int(num[target])-1]
        else:
            unit = labl['unit'][int(num[target])-1]
            if int(num[target-1]) > 4:
                decimals = f"{labl['prefix'][int(num[target-1])-5] + labl['suffix'][1]} {unit}"
            else:
                decimals = f"{labl['decim'][int(num[target-1])-2] } {unit}"

    if len(num) == 3:
        starter = f"{labl['unit'][int(num[0])-1]} hundred"
        return f"{starter} {decimals}"
    elif len(num) == 2:
        return decimals
    elif len(num) == 1:
        return labl['unit'][int(num[0])-1]

def main():
    base = input('Enter a number (less than 999): ')
    print(wordnumber(base))
if __name__ == '__main__':
    main()

File: projects/m2/020-write-out-numbers-in-english/solution/giannighidoni/solution-write-out-numbers-in-english.js
const inputNumber = 369;
const inputNumberString = String(inputNumber);

const number = {
    1 : "one",
    2 : "two",
    3 : "three",
    4 : "four",
    5 : "five",
    6 : "six",
    7 : "seven",
    8 : "eight",
    9 : "nine"
};
const teen = {
    0 : "ten",
    1 : "eleven",
    2 : "twelve",
    3 : "thirteen",
    4 : "fourteen",
    5 : "fifteen",
    6 : "sixteen",
    7 : "seventeen",
    8 : "eighteen",
    9 : "nineteen"
};
const tens = {
    2 : "twenty",
    3 : "thirty",
    4 : "forty",
    5 : "fifty",
    6 : "sixty",
    7 : "seventy",
    8 : "eighty",
    9 : "ninety"
};
const outputArr = [];

switch(inputNumberString.length){
    case 3 :
        for(let key in number){
            if(inputNumberString[0] == key){
                outputArr.push(number[key]);
                outputArr.push("hundred");
            }
        }
        if(inputNumberString[1] == 0){
            for(let key in number){
                if(inputNumberString[2] === key){
                    outputArr.push(number[key]);
                }
            }
        } else if(inputNumberString[1] == 1){
            for(let key in teen){
                if(inputNumberString[2] == key){
                    outputArr.push(teen[key]);
                }
            }
        } else if(inputNumberString[1] > 1){
            for(let key in tens){
                if(inputNumberString[1] == key){
                    outputArr.push(tens[key]);
                }
            }
            for(let key in number){
                if(inputNumberString[2] == key){
                    outputArr.push(number[key]);
                }
            }
        }
        break;
    case 2 :
        if(inputNumber[0] === 1){
            for(let key in teen){
                if(inputNumber[1] == key){
                    outputArr.push(teen[key]);
                }
            }
        } else {
            for(let key in tens){
                if(inputNumberString[0] == key){
                    outputArr.push(tens[key]);
                }
            }
            for(let key in number){
                if(inputNumberString[1] == key){
                    outputArr.push(number[key]);
                }
            }
        }
        break;
    case 1 :
        for(let key in number){
            if(inputNumberString[0] == key){
                outputArr.push(number[key]);
            }
        }
        break;
}

const outResult = outputArr.join(" ");
console.log(outResult);